tests/QuikGraph.Tests/Algorithms/Contracts/DistancesCollectionContract.cs

```diff
@@ -0,0 +1,304 @@
+using System;
```

Comment:
To make it easier for me to name the tests I have created subclasses for each method tested. I opted for placing all the classes within the same file. Would you rather have them be split out into separate files and then put everything into its own directory, or is this okay?

Reply:
For now we can keep them in the same folder. We will see if we have a lot of contracts we will maybe split in sub folders. For now I just split in several files.